lib/_stream_readable.js

```diff
@@ -853,7 +853,15 @@ Readable.prototype.removeAllListeners = function(ev) {
 };
 
 function updateReadableListening(self) {
-  self._readableState.readableListening = self.listenerCount('readable') > 0;
+  const state = self._readableState;
+  state.readableListening = self.listenerCount('readable') > 0;
+  // try to start flowing for 'data' listeners
+  // (if pipesCount is not 0 then we have already 'flowed')
+  if (!state.readableListening &&
+      self.listenerCount('data') > 0 &&
+      state.pipesCount === 0) {
```

Comment:
Do we have a test covering what happens if the last `readable` listener is removed but there is a pipe?

Reply:
Added.